app/backend/app.py

```diff
@@ -84,10 +84,26 @@ async def assets(path):
 
 
 # Serve content files from blob storage from within the app to keep the example self-contained.
-# *** NOTE *** this assumes that the content files are public, or at least that all users of the app
-# can access all the files. This is also slow and memory hungry.
+# *** NOTE *** if AZURE_ENFORCE_ACCESS_CONTROL is not set or false, all content is accessible
+# This is also slow and memory hungry.
 @bp.route("/content/<path>")
 async def content_file(path: str):
+    # If authentication is enabled, validate the user can access the file
```

Comment:
It could be nice design wise if this was a decorator, like @access_control required, but the path-checking part is specific to this route, so am not sure if decorator would be reusable in other scenarios. Typically in Python apps, I've used decorator's to denote different kinds of auth situations, and then it's easy to attach them to various routes (auth_required, admin_required, etc). It would also make the code cleaner for developers not using authentication. What do you think, would that be a useful pattern here?

Reply:
Yes, I'll take a look at this pattern